pkg/scheduler/cache/status_updater/concurrency.go

```diff
@@ -30,24 +30,56 @@ func (su *defaultStatusUpdater) SyncPodGroupsWithPendingUpdates(podGroups []*eng
 	for i := range podGroups {
 		key := su.keyForPodGroupPayload(podGroups[i].Name, podGroups[i].Namespace, podGroups[i].UID)
 		usedKeys[key] = true
-		inflightUpdateAny, found := su.inFlightPodGroups.Load(key)
-		if !found {
+		pgLatestUpdate, inFlightUpdatefound, appliedUpdateFound := su.getLatestPgUpdate(key)
+		if !inFlightUpdatefound && !appliedUpdateFound {
 			continue
 		}
-		podGroup := inflightUpdateAny.(*inflightUpdate).object.(*enginev2alpha2.PodGroup)
-		isPodGroupUpdated := su.syncPodGroup(podGroup, podGroups[i])
-		if isPodGroupUpdated {
-			su.inFlightPodGroups.Delete(key)
+		podGroup := pgLatestUpdate.object.(*enginev2alpha2.PodGroup)
+		podGroupMatchesUpdate := su.syncPodGroup(podGroup, podGroups[i])
+		// Delete the inflight update if it was applied + the pod group in the lister matches the inFlight
+		if podGroupMatchesUpdate {
+			su.cleanPgUpdatesCaching(key, inFlightUpdatefound, appliedUpdateFound)
```

Comment:
This function will delete it even from the `inFlightPodGroups` meaning pod groups that were not updated yet. I think it will not fix the bug.

Reply:
Fixed